stmhal/usbd_conf.c

```diff
@@ -379,90 +383,97 @@ void HAL_PCD_DisconnectCallback(PCD_HandleTypeDef *hpcd)
 USBD_StatusTypeDef  USBD_LL_Init (USBD_HandleTypeDef *pdev)
 { 
 #if defined(USE_USB_FS)
+if (pdev->id == 0 )

```

Comment:
In the init code the hardware base registers USB_OTG_FS or USB_OTG_HS are not set up, and its purely a logical assignment (it is the way the Cube_FW_V4 example code does it - not that it shouldn't be improved on where ever possible - and I've created an enum.)
 I have changed it to be an enum to be more clear.
For (USBD_HandleTypeDef *pdev->id == USB_PHY_FS_ID)
and it then sets-up   
(PCD_HandleTypeDef )pcd_Xs_handle.Instance = USB_OTG_FS; 
and similarly for  base hw register USB_OTG_HS;
Happy to do it another way if you can see it, it is initiated in 
usb.c: USBD_Init(&hUSBDDevice, (USBD_DescriptorsTypeDef*)&USBD_Descriptors, USB_PHY_FS_ID);